### PR TITLE
Column families for temporal indexes

### PR DESCRIPTION
**Pull request by jonpither**
Sets up column families for temporal indexes. Minimizes temporal seek time during ingest.

This could be seen as a part 1 of column families. We could bring in more CFs and measure the respective performance gains. Note that each CF brings in a 64 mb by default mem-table.

Rebased on #1762

### CODE REVIEW

File: modules/rocksdb/src/xtdb/rocksdb.clj [hunk 1]
```diff
@@ -13,65 +13,98 @@
...
+                        WriteBatch WriteOptions Statistics StatsLevel
+                        ColumnFamilyOptions ColumnFamilyDescriptor ColumnFamilyHandle)))
+
+(defprotocol CfId
+  (->cf-id [this]))
+
+(extend-protocol CfId
+  (class (byte-array 0))
+  (->cf-id [this]
+    (first this))
```
Comment (jarohen) [inline modules/rocksdb/src/xtdb/rocksdb.clj]: `(aget this 0)`, save going via Clojure's `seq` abstraction

File: modules/rocksdb/src/xtdb/rocksdb.clj [hunk 2]
```diff
@@ -13,65 +13,98 @@
...
 
+(def ^:const column-family-defs [c/entity+vt+tt+tx-id->content-hash-index-id
+                                 c/entity+z+tx-id->content-hash-index-id])
+
 (defn- iterator->key [^RocksIterator i]
   (when (.isValid i)
     (mem/as-buffer (.key i))))
 
-(defrecord RocksKvIterator [^RocksIterator i]
+(defrecord RocksKvIterator [i is k->i]
```
Comment (jarohen) [inline modules/rocksdb/src/xtdb/rocksdb.clj]: given this is low-level, I'd go for some lower-level, more performant primitives here:
* `defrecord` -> `deftype` (so you can use mutable fields)
* `is`: `(atom {})` -> `(HashMap.)`
* `i`: `(atom nil)` -> `^:unsynchronised-mutable ^RocksIterator i`

File: modules/rocksdb/src/xtdb/rocksdb.clj [hunk 3]
```diff
@@ -13,65 +13,98 @@
...
+    (iterator->key ^RocksIterator @i))
 
   (value [this]
-    (mem/as-buffer (.value i)))
+    (mem/as-buffer (.value ^RocksIterator @i)))
 
   Closeable
   (close [this]
-    (.close i)))
+    (doseq [[_ i] @is]
```
Comment (jarohen) [inline modules/rocksdb/src/xtdb/rocksdb.clj]: `(doseq [i (.values is)] ...)`

**Comment by jonpither** [comment]
Changes made @jarohen. We have also made the generative test run for non-rocks implementations still.

**Comment by jonpither** [comment]
There's an issue with entity history `asc` - it's broken. Tests were not picked up earlier, since fork-test / api-test only ran against the in-mem DB.

**Comment by jonpither** [comment]
Made column families configurable as per @hraberg request.

**Pull request merged**